src/app/components/icon/icon.ts

```diff
@@ -0,0 +1,38 @@
+import { NgClass, NgComponentOutlet, NgStyle } from '@angular/common';
+import { ChangeDetectionStrategy, Component, InjectionToken, Input, Type, inject } from '@angular/core';
+
+export interface PIconComponent {
+    icon: string;
+    iconStyle?: NgStyle['ngStyle'];
+}
+
+export const P_ICON_COMPONENT = new InjectionToken<Type<PIconComponent>>('P_ICON_COMPONENT');
+
+/**
+ * Icon is a simple component to render an icon.
+ * @group Components
+ */
+@Component({
+    standalone: true,
+    imports: [NgClass, NgStyle, NgComponentOutlet],
+    selector: 'p-icon',
+    template: `
+        @if(!iconComponentClass) {
+        <i [ngClass]="icon" [ngStyle]="iconStyle"></i>
```

Comment:
ngClass & ngStyle might be deprecated, [class] & [style] are preferred.

Reply:
Do you have a link to support this claim that they are deprecated? The official docs don't say anything about that fact https://angular.dev/api/common/NgClass

Reply:
sure, see this for details:
https://github.com/angular/angular/pull/58860
its also mentioned in upcoming style guide.

Reply:
Interesting thanks. I will change it.